### ISSUE TITLE
POSIT Connect server incompatible with versions gt v0.8.0

### ISSUE THREAD

**Issue by mccauleyp** [posit-dev/py-shiny#1234]
I have a simple Shiny express app that renders some tables. It works locally with `shiny run` as expected on versions `shiny==0.8.0` and `0.8.1`, but those versions break the app when deployed to my company's internal POSIT Connect server, for which I need to roll back to `shiny==0.7.1`. I deploy the app using `rsconnect deploy shiny` with all the default settings. 

The deployment succeeds and page loads, but the rendering fails and looks like the following: 

![Screenshot 2024-03-20 at 2 51 29 PM](https://github.com/posit-dev/py-shiny/assets/66882763/4239c312-9125-43e1-86c7-d5ea60c672ff)

The startup logs look like:

```
2024/03/20 2:51:57 PM: Linux distribution: Red Hat Enterprise Linux release 8.9 (Ootpa)
2024/03/20 2:51:57 PM: Running as user: uid=983(rstudio-connect) gid=983(rstudio-connect) groups=983(rstudio-connect)
2024/03/20 2:51:57 PM: Connect version: 2023.12.0
2024/03/20 2:51:57 PM: LANG: en_US.UTF-8
2024/03/20 2:51:57 PM: Working directory: /opt/rstudio-connect/mnt/app
2024/03/20 2:51:57 PM: Bootstrapping environment using Python 3.11.3 (main, Jun  4 2023, 22:35:24) [GCC 8.5.0 20210514 (Red Hat 8.5.0-16)] at /opt/python/3.11.3/bin/python3.11
2024/03/20 2:51:57 PM: Running content with the Python virtual environment /opt/rstudio-connect/mnt/app/python/env (/opt/rstudio-connect/mnt/python-environments/pip/3.11.3/A-Ec3OMeEboTM3ULs1TTfw)
2024/03/20 2:51:57 PM: Running content using Python "3.11.3 (main, Jun  4 2023, 22:35:24) [GCC 8.5.0 20210514 (Red Hat 8.5.0-16)]" at "/opt/rstudio-connect/mnt/app/python/env/bin/python"
2024/03/20 2:51:57 PM: Loading code from "shiny.express.app:app_2e_py"
2024/03/20 2:51:58 PM: App type is "App"
2024/03/20 2:51:58 PM: Starting server...
```

The code looks like the following:

```
from shiny import reactive
from shiny.express import input, render, ui

with ui.navset_pill(id="tab"):
    with ui.nav_panel("Study Metadata"):

        @render.data_frame
        def study_metadata() -> render.DataGrid:
            ...
``` 
Any ideas what might be going on?

**Comment by wch** [comment]
Can you try installing the development version of rsconnect-python and see if that fixes your issue?

```
pip install git+https://github.com/rstudio/rsconnect-python.git#egg=rsconnect-python
```

I think this may have been fixed in https://github.com/rstudio/rsconnect-python/pull/545, but that fix has not yet been released.

**Comment by mccauleyp** [comment]
> Can you try installing the development version of rsconnect-python and see if that fixes your issue?
> 
> ```
> pip install git+https://github.com/rstudio/rsconnect-python.git#egg=rsconnect-python
> ```
> 
> I think this may have been fixed in [rstudio/rsconnect-python#545](https://github.com/rstudio/rsconnect-python/pull/545), but that fix has not yet been released.

Yep, that did it! Thanks!
